pysoa/common/transport/redis_gateway/backend/base.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import, unicode_literals
+
+import binascii
+import itertools
+import random
+
+import six
+
+
+class LuaRedisCommand(object):
+    _script = ''
+
+    def __init__(self, redis_connection):
+        """
+        Registers this Lua script with the connections. The supplied redis connection is only used to create a
+        registered script option; no connection is established or communicated over.
+
+        :param redis_connection: The connection for registering this script
+        """
+        self._redis_script = redis_connection.register_script(self._script.strip())
+
+    def _call(self, keys, args, connection):
+        return self._redis_script(keys=keys, args=args, client=connection)
+
+
+class SendMessageToQueueCommand(LuaRedisCommand):
+    # KEYS[1] = queue key
+    # ARGV[1] = expiry
+    # ARGV[2] = queue capacity
+    # ARGV[3] = message
+    _script = """
+if redis.call('llen', KEYS[1]) >= tonumber(ARGV[2]) then
+    return redis.error_reply("queue full")
+end
+redis.call('rpush', KEYS[1], ARGV[3])
+redis.call('expire', KEYS[1], ARGV[1])
+"""
+
+    def __call__(self, queue_key, message, expiry, capacity, connection):
+        self._call(keys=[queue_key], args=[expiry, capacity, message], connection=connection)
+
+
+class BaseRedisClient(object):
+    DEFAULT_RECEIVE_TIMEOUT = 5
+    RESPONSE_QUEUE_SPECIFIER = '!'
+
+    def __init__(self, ring_size):
+        self._ring_size = ring_size
+        self._receive_index_generator = itertools.cycle(range(self._ring_size))  # may be overridden by subclasses
```

Comment:
`_receive_index_generator` is never used.

Reply:
Good catch. Fixed.